src/nycdb/database.py

```diff
@@ -1,64 +1,53 @@
 import os
-import psycopg2
-import psycopg2.extras
+import psycopg
 from . import sql
 
 
 class Database:
     """
-    Database connection. This is a wrapper around psycopg2.connection, accessible at `.conn`
-    see http://initd.org/psycopg/docs/connection.html for psycopg2's documentation
+    Database connection. This is a wrapper around psycopg.connection, accessible at `.conn`
+    see http://initd.org/psycopg/docs/connection.html for psycopg's documentation
 
 
     """
+
     def __init__(self, args, table_name=None):
         """
         args is a Namespace that must contain: user, password, database, host, and port
         """
-        self.conn = psycopg2.connect(
-            user=args.user,
-            password=args.password,
-            host=args.host,
-            database=args.database,
-            port=args.port
-        )
-
-        self.table_name = table_name
-
         self.connection_params = {
-            'user': args.user,
-            'password': args.password,
-            'host': args.host,
-            'database': args.database,
-            'port': args.port
+            "user": args.user,
+            "password": args.password,
+            "host": args.host,
+            "database": args.database,
+            "port": args.port,
         }
 
+        self.table_name = table_name
+        self.conn = psycopg.connect(self.conninfo())
+
     def sql(self, SQL):
-        """ executes single sql statement """
+        """executes single sql statement"""
         with self.conn.cursor() as curs:
             curs.execute(SQL)
         self.conn.commit()
 
     def insert_rows(self, rows, table_name=None):
         """
-        Inserts many rows, all in the same transaction, using psycopg2.extras.execute_values
+        Inserts many rows, all in the same transaction using executemany
         """
 
         if table_name is None:
             table_name = self.table_name
 
         with self.conn.cursor() as curs:
-            sql_str, template = sql.insert_many(table_name, rows)
             try:
-                psycopg2.extras.execute_values(
-                    curs,
-                    sql_str,
-                    rows,
-                    template=template,
-                    page_size=len(rows)
+                curs.executemany(
+                    sql.insert_many(table_name, rows),
+                    map(lambda d: list(d.values()), rows),
```

Comment:
@aepyornis hey, I'm just now catching up with these changes for our auto-updater, and I'm noticing that everything is considerably slower now. For most datasets it isn't a big deal, but for the really large ones (pluto, acris) it is often timing out and fails to actually update. I looked through and this is the only place I could see that might impact performance. It seems like the psycopg2.extensions `execute_values` and `execute_batch` don't exists for 3. It says the new `executemany` should be faster than it used to be, but maybe is still slower than the old extensions methods. It says there are now ways to use `copy` that might be fast. Did you look into that at all, or notice any performance issues when you were making these changes?

Reply:
Interesting -- let's investigate why it's slower. On my local nycdb instance I didn't notice any major speed differences either way. I can't recall why `list(d.values())` was necessary but perhaps the slowdown is the extra list object created? just I guess I have no idea. What version of python are you using? 

Years ago when I looked into using COPY it didn't seem compatible with our setup. We process the CSVs in python line-by-line first to add columns, perform our own type-casting and error checking, but maybe we can revisit COPY and see what's possible.

Reply:
I think I figured out what the issue was, and have an idea for a solution. Going to test it out a bit more, but curious what you think #266